src/test/java/jenkins/scm/impl/subversion/SubversionSCMFileSystemTest.java

```diff
@@ -0,0 +1,189 @@
+		long trunk = sampleRepo.revision();
+		assertEquals(3, trunk);
+		sampleRepo.svnkit("copy", "--message=branching", sampleRepo.trunkUrl(), sampleRepo.branchesUrl() + "/dev");
+		sampleRepo.svnkit("switch", sampleRepo.branchesUrl() + "/dev", sampleRepo.wc());
+		sampleRepo.write("file", "dev1");
+		sampleRepo.svnkit("commit", "--message=dev1", sampleRepo.wc());
+		long dev1 = sampleRepo.revision();
+		assertEquals(5, dev1);
+		sampleRepo.svnkit("copy", "--message=tagging", sampleRepo.branchesUrl() + "/dev",
+				sampleRepo.tagsUrl() + "/dev-1");
+		sampleRepo.write("file", "dev2");
+		sampleRepo.svnkit("commit", "--message=dev2", sampleRepo.wc());
+		long dev2 = sampleRepo.revision();
+		assertEquals(7, dev2);
+		SCMSource source = new SubversionSCMSource(null, sampleRepo.prjUrl());
+		try (SCMFileSystem fs = SCMFileSystem.of(source, new SCMHead("trunk"))) {
+			assertThat(fs, notNullValue());
+			SCMFile root = fs.getRoot();
+			assertThat(root, notNullValue());
+			assertTrue(root.isRoot());
+			Iterable<SCMFile> children = root.children();
+			Iterator<SCMFile> iterator = children.iterator();
+			assertThat(iterator.hasNext(), is(true));
+			SCMFile file = iterator.next();
+			assertThat(iterator.hasNext(), is(false));
+			assertThat(file.getName(), is("file"));
+		}
+	}
+
+	@Test
+	public void ofSourceRevision() throws Exception {
+		sampleRepo.init();
+		sampleRepo.svnkit("copy", "--message=branching", sampleRepo.trunkUrl(), sampleRepo.branchesUrl() + "/dev");
+		sampleRepo.svnkit("switch", sampleRepo.branchesUrl() + "/dev", sampleRepo.wc());
+		SCMSource source = new SubversionSCMSource(null, sampleRepo.prjUrl());
+		SCMRevision revision = source.fetch(new SCMHead("branches/dev"), null);
+		sampleRepo.write("file", "modified");
+		sampleRepo.svnkit("commit", "--message=dev1", sampleRepo.wc());
+		try (SCMFileSystem fs = SCMFileSystem.of(source, new SCMHead("branches/dev"), revision)) {
+			assertThat(fs, notNullValue());
+			SCMFile root = fs.getRoot();
+			assertThat(root, notNullValue());
+			Iterable<SCMFile> children = root.children();
+			Iterator<SCMFile> iterator = children.iterator();
+			assertThat(iterator.hasNext(), is(true));
+			SCMFile file = iterator.next();
+			assertThat(iterator.hasNext(), is(false));
+			assertThat(file.getName(), is("file"));
+			assertThat(file.contentAsString(), is(""));
+		}
+	}
+
+	@Test
+	public void lastModified_Smokes() throws Exception {
+		sampleRepo.init();
+		sampleRepo.svnkit("copy", "--message=branching", sampleRepo.trunkUrl(), sampleRepo.branchesUrl() + "/dev");
+		sampleRepo.svnkit("switch", sampleRepo.branchesUrl() + "/dev", sampleRepo.wc());
+		SCMSource source = new SubversionSCMSource(null, sampleRepo.prjUrl());
+		SCMRevision revision = source.fetch(new SCMHead("branches/dev"), null);
+		sampleRepo.write("file", "modified");
+		sampleRepo.svnkit("commit", "--message=dev1", sampleRepo.wc());
+		final long fileSystemAllowedOffset = isWindows() ? 4000 : 1500;
```

Comment:
This offset probably needs to be increased for this test to pass on the build server. It seems that 1500ms is about 200ms too few for it to pass.

Reply:
Not sure what the function of the code is, but if it has to check if the file was generated after the action started, it would be better to populate currentTime before the action is executed and forget the offset. This looks more like a performance test.